### PR TITLE
Reuse the wizard's check-answers permissions list on the Organisation edit page

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

The "show permissions" page accessible via Organisations > Organisation name is different from the prototype and from the Check answers page in the wizard. There's no need for two different implementations of this permissions list and the other implementation is better, so standardise on that.

## Changes proposed in this pull request

**The old version**

Screenshot 2020-09-02 at 11 12 15

**The new version**

Screenshot 2020-09-02 at 11 14 20

The problem with using the wizard's component is that it depends on the wizard!

The first job is to break that dependency, which I've done by having it work on `ProviderRelationshipPermissions` models. I then changed the wizard to be able to emit those without the help of `SetupProviderRelationshipPermissions`. It used to be the case that that class took in the wizard's data and turned it into `ProviderRelationshipPermissions`. Now neither `SetupProviderRelationshipPermissions` nor the view component have any idea whether they're being fed by the wizard or by something else.

Then it was necessary to make the `Change` links work in both contexts. To do this I made a couple of change link builders (ick?) and passed them in.

Finally the component needed to decide whether or not to show the change links (those shown on the Organisations pages are not necessarily editable), so I added a final `editable` parameter.

## Guidance to review

The multiple change links should have correct visually-hidden text and should link to the checkboxes.

The wizard should continue to work.

## Link to Trello card

https://trello.com/c/6ZwbrG2o/2706-bring-the-organisations-edit-page-into-parity-with-the-wizards-check-answers-page

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/provider_interface/setup_provider_relationship_permissions.rb [hunk 1]
```diff
@@ -2,27 +2,17 @@ module ProviderInterface
...
   class SetupProviderRelationshipPermissions
-    def self.call(permissions_data = {})
+    def self.call(models)
       ProviderRelationshipPermissions.transaction do
-        permissions_data.each do |id, permissions|
-          record = ProviderRelationshipPermissions.find(id)
+        models.map do |record|
           raise PermissionsSetupError, 'Permissions record has already been setup' if record.setup_at.present?
 
-          record.update!(permissions_attributes_for_persistence(permissions))
```
Comment (steventux) [inline app/services/provider_interface/setup_provider_relationship_permissions.rb:11]: Does `save!`  provider the same audit trail info about attributes? I'm gonna assume it does (the docs don't seem to show an example of this).

File: app/components/provider_interface/provider_relationship_permissions_list_component.html.erb [hunk 2]
```diff
@@ -5,17 +5,19 @@
       <%= row[:key] %>
     </dt>
     <dd class="govuk-summary-list__value">
-      <ul class="govuk-list make-decisions">
+    <ul class="govuk-list <%= row[:permission_slug] %>">
```
Comment (steventux) [inline app/components/provider_interface/provider_relationship_permissions_list_component.html.erb:8]: Needs indentation.

File: app/forms/provider_interface/provider_relationship_permissions_setup_wizard.rb [hunk 3]
```diff
@@ -37,6 +37,28 @@ def permissions_for_relationship(relationship_id = current_provider_relationship
       provider_relationship_permissions.fetch(relationship_id.to_s, {})
     end
 
+    # We hold data in the form
+    # { permission_id => { permission1 => [ "training" ], permission2 => [ "training", "ratifying" ] } }
+    #
+    # This method massages it into attrs suitable for ProviderRelationshipPermissions models
+    # and returns those models ready for persistence, either for actually saving or for previewing
+    # on the check-answers page
+    def permissions_for_persistence
```
Comment (steventux) [inline app/forms/provider_interface/provider_relationship_permissions_setup_wizard.rb:46]: :+1: To this refactor, much better placed here.

**Review by steventux — APPROVED** [review]
This looks good to me, I've added a question about audit entries more out of ignorance than anything else.

File: app/services/provider_interface/setup_provider_relationship_permissions.rb (hunk 1 above)
Comment (duncanjbrown) [reply app/services/provider_interface/setup_provider_relationship_permissions.rb:11]: Good question, checking this...

Comment (duncanjbrown) [reply app/services/provider_interface/setup_provider_relationship_permissions.rb:11]: Looks OK:

Screenshot 2020-09-07 at 16 23 38

**Pull request merged**